Using backend: pytorch
WARNING:root:The OGB package is out of date. Your version is 1.2.5, while the latest version is 1.3.3.
Namespace(seed=2, dataset='ogbn-mag', gpu=0, root='../data/', emb_path='../data/', stages=[400, 400, 400, 400], extra_embedding='complex', embed_size=256, num_hops=2, label_feats=True, num_label_hops=2, hidden=512, dropout=0.5, n_layers_1=2, n_layers_2=2, n_layers_3=4, input_drop=0.1, att_drop=0.0, label_drop=0.0, residual=True, act='leaky_relu', bns=True, label_bns=True, amp=True, lr=0.001, weight_decay=0.0, eval_every=1, batch_size=10000, patience=100, threshold=0.75, gama=10.0, start_stage=0, reload='', moving_k=1, store_model=False, moe_widget=['feat_ngnn', 'label_ngnn'], num_expert=4, top_k=1, expert_drop=0.0, gate='naive')
Use extra embeddings generated with the complex method
Current num hops = 2
./output/ogbn-mag/17de274074644e5da8a5f87c404e2ffe 

Stage 0 

Current num label hops = 2
# Params: 9423907
Epoch:20, avg epoch time:19.6312
 Train loss:1.5984238348309956, Val loss:1.6359057426452637, Test loss:1.6737620830535889
 Train acc:55.7853, Val acc:53.1852, Test acc:51.8038
Best Epoch 18,Val 53.5119, Test 51.9397

Epoch:40, avg epoch time:19.3748
 Train loss:1.4694954611006237, Val loss:1.563361406326294, Test loss:1.6029047966003418
 Train acc:58.0568, Val acc:54.5647, Test acc:53.0556
Best Epoch 39,Val 54.5647, Test 53.0556

Epoch:60, avg epoch time:19.3665
 Train loss:1.4032656805855888, Val loss:1.5632479190826416, Test loss:1.6019469499588013
 Train acc:59.2727, Val acc:54.1824, Test acc:52.7170
Best Epoch 53,Val 55.1349, Test 53.6160

Traceback (most recent call last):
  File "/data/liweikai/sehgnn/large/main.py", line 407, in <module>
    main(args)
  File "/data/liweikai/sehgnn/large/main.py", line 265, in main
    loss_train, train_acc = train(model, train_loader, loss_fcn, optimizer, evaluator, device, feats, label_feats, labels_cuda, label_emb, scalar=scalar)
  File "/data/liweikai/sehgnn/large/utils.py", line 138, in train
    for batch in train_loader:
  File "/home/cenyukuo/anaconda3/envs/drgat/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 521, in __next__
    data = self._next_data()
  File "/home/cenyukuo/anaconda3/envs/drgat/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 561, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/cenyukuo/anaconda3/envs/drgat/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 49, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/cenyukuo/anaconda3/envs/drgat/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 49, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
KeyboardInterrupt
